romancal/tweakreg/tweakreg_step.py

```diff
@@ -528,7 +528,6 @@ def _common_name(group):
     for im in group:
         im = im if isinstance(im, datamodels.DataModel) else datamodels.open(im)
         file_names.append(path.splitext(im.meta.filename)[0].strip('_- '))
```

Comment:
What about something like this instead?
```suggestion
        if isinstance(im, datamodels.DataModel):
            fn = im.meta.filename
        else:
            with datamodels.open(im) as dm:
                fn = dm.meta.filename
        file_names.append(path.splitext(fn)[0].strip('_- '))
```
This should handle both cases:
- when `im` is a data model, don't close the model
- when `im` is a filename, open, and then close the data model (using a with context)

Reply:
I like Brett's version.